src/io/flutter/run/daemon/DeviceSelection.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright  2017 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.run.daemon;
+
+import com.google.common.collect.ImmutableList;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+
+/**
+ * An immutable snapshot of the list of connected devices and current selection.
+ */
+class DeviceSelection {
+  static final DeviceSelection EMPTY = new DeviceSelection(ImmutableList.of(), null);
+
+  private final ImmutableList<FlutterDevice> devices;
+  private final @Nullable FlutterDevice selection;
+
+  private DeviceSelection(ImmutableList<FlutterDevice> devices, @Nullable FlutterDevice selected) {
+    this.devices = devices;
+    this.selection = selected;
+  }
+
+  @NotNull ImmutableList<FlutterDevice> getDevices() {
```

Comment:
How are we ensuring that this is not null?  It seems like `devices` the field (or constructor arg) should be annotated.  Or?